core/chains/evm/txmgr/finalizer.go

```diff
@@ -0,0 +1,294 @@
+package txmgr
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/mailbox"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+var _ Finalizer = (*evmFinalizer)(nil)
+
+// processHeadTimeout represents a sanity limit on how long ProcessHead should take to complete
+const processHeadTimeout = 10 * time.Minute
+
+type finalizerTxStore interface {
+	FindConfirmedTxesReceipts(ctx context.Context, finalizedBlockNum int64, chainID *big.Int) ([]Receipt, error)
+	UpdateTxStatesToFinalizedUsingReceiptIds(ctx context.Context, txs []int64, chainId *big.Int) error
+}
+
+type finalizerChainClient interface {
+	BatchCallContext(ctx context.Context, elems []rpc.BatchElem) error
+}
+
+type finalizerHeadTracker interface {
```

Comment:
I still curious why do we need to create an private interface here ? Can we import `httypes.HeadTracker`. Cuz I saw this [PR](https://github.com/smartcontractkit/chainlink/pull/13973/files) did the same. It would be a lot duplicate code every time we use headTracker elsewhere.

Reply:
It's a standard we're trying to follow moving forward so you'll definitely see places around the code base that don't follow this. The reason for this though is so everything is self contained in the component's file and drilled down to only what is required.

Reply:
I found there are couple other places contain usage of of `HeaderTracker`, all of them are using imported `logpoller.HeadTracker` instead of making a self-contain interface, including ([chainReader](https://github.com/smartcontractkit/chainlink/blob/15dc74cabd3a83041ca97df54ea0fbb7e76e2a0a/core/services/relay/evm/chain_reader.go#L36), [methodBinding](https://github.com/smartcontractkit/chainlink/blob/15dc74cabd3a83041ca97df54ea0fbb7e76e2a0a/core/services/relay/evm/method_binding.go#L31)). 

IMHO, I prefer to have the consistency for the finality [PR](https://github.com/smartcontractkit/chainlink/pull/13973)

Reply:
I think those components are outside of the TXM so it won't necessarily be inconsistent if we do something differently for this PR. If you have concerns though, you can update the PR the way you were thinking and maybe we can discuss during its review.